### PR TITLE
Feature/flickr photos vc part2

### PR DESCRIPTION
**Pull request by codesinc**
This PR includes part 2 of Flickr

### CODE REVIEW

**Comment by CodeBaseCamp** [comment]
Please use reasonable commit titles and descriptions (`same` isn't one of them :) ).

File: FlickrPhotos/FlickrPhotos.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -7,21 +7,25 @@
 	objects = {
 
 /* Begin PBXBuildFile section */
+		C930DBF11CCFA6E9001BEE3E /* PlaceListViewController.m in Sources */ = {isa = PBXBuildFile; fileRef = C930DBF01CCFA6E9001BEE3E /* PlaceListViewController.m */; };
+		C930DBF51CCFA725001BEE3E /* PhotoListViewController.m in Sources */ = {isa = PBXBuildFile; fileRef = C930DBF41CCFA725001BEE3E /* PhotoListViewController.m */; };
 		C93264BB1CBBD92B00B6F4C0 /* FlickrMetadataDownloader.m in Sources */ = {isa = PBXBuildFile; fileRef = C93264AF1CBBD92B00B6F4C0 /* FlickrMetadataDownloader.m */; };
 		C93264BC1CBBD92B00B6F4C0 /* FlickrFetcher.m in Sources */ = {isa = PBXBuildFile; fileRef = C93264B31CBBD92B00B6F4C0 /* FlickrFetcher.m */; };
 		C93264BD1CBBD92B00B6F4C0 /* FlickrPhotoMetadata.m in Sources */ = {isa = PBXBuildFile; fileRef = C93264B61CBBD92B00B6F4C0 /* FlickrPhotoMetadata.m */; };
-		C93264BE1CBBD92B00B6F4C0 /* FlickrPlacePhotoMetadata.m in Sources */ = {isa = PBXBuildFile; fileRef = C93264B91CBBD92B00B6F4C0 /* FlickrPlacePhotoMetadata.m */; };
```
Comment (CodeBaseCamp) [inline FlickrPhotos/FlickrPhotos.xcodeproj/project.pbxproj]: Why are files getting removed in this PR?

File: FlickrPhotos/FlickrPhotos/Base.lproj/Main.storyboard [hunk 2]
```diff
@@ -1,48 +0,0 @@
-<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Comment (CodeBaseCamp) [inline FlickrPhotos/FlickrPhotos/Base.lproj/Main.storyboard]: Didn't we agree that these changes should be added in separate PRs?

File: FlickrPhotos/FlickrPhotos/ImageViewController.h [hunk 3]
```diff
@@ -0,0 +1,18 @@
+// Copyright (c) 2016 Lightricks. All rights reserved.
+// Created by codesinc.
+
+#import <UIKit/UIKit.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@class FlickrPhotoMetadata;
+
+/// View controller that updates its image view with a given image in Flickr.
```
Comment (CodeBaseCamp) [inline FlickrPhotos/FlickrPhotos/ImageViewController.h]: What does `image in Flickr` mean? A random image retrieved from Flickr? The image of the year? Please make sure to give insightful documentations.

File: FlickrPhotos/FlickrPhotos/ImageViewController.h [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+@interface ImageViewController : UIViewController
+
+/// Image metadata depicted by this instance.
```
Comment (CodeBaseCamp) [inline FlickrPhotos/FlickrPhotos/ImageViewController.h]: `Metadata of image depicted by this instance.` (otherwise, the metadata is depicted rather than the image).

File: FlickrPhotos/FlickrPhotos/ImageViewController.m [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+  
+  PhotoCompletionBlock onComplete = ^(UIImage *image) {
+    dispatch_async(dispatch_get_main_queue(), ^{
+      [self.spinner stopAnimating];
+      self.image = image;
+    });
+  };
+  
+  FlickrPhotoContentDownloader *downloader;
+  downloader = [[FlickrPhotoContentDownloader alloc] init];
```
Comment (CodeBaseCamp) [inline FlickrPhotos/FlickrPhotos/ImageViewController.m]: Collapse lines 72 and 73 to a single line.

File: FlickrPhotos/FlickrPhotos/Model/Downloader/FlickrPhotoMetadataDownloader.m [hunk 6]
```diff
@@ -19,32 +19,38 @@ @interface FlickrPhotoMetadataDownloader()
...
       return;
     }
-    
-    NSMutableArray<FlickrPhotoMetadata *> *photosMetadata =
-    [[NSMutableArray<FlickrPhotoMetadata *> alloc] initWithCapacity:photos.count];
-    for (NSDictionary *metadata in photos) {
-      NSURL *photoURL =
-          [FlickrFetcher URLforPhoto:metadata format:FlickrPhotoFormatLarge];
-      [photosMetadata addObject:[[FlickrPhotoMetadata alloc] initWithUrl:photoURL]];
+    NSMutableArray<FlickrPhotoMetadata *> *photoList =
```
Comment (CodeBaseCamp) [inline FlickrPhotos/FlickrPhotos/Model/Downloader/FlickrPhotoMetadataDownloader.m]: `photoList` suggests its a list of photos. However, the array contains metadata. This should be reflected by the array name. I'd go with a plain `metadata`.

File: FlickrPhotos/FlickrPhotos/Model/Downloader/FlickrPhotoMetadataDownloader.m [hunk 7]
```diff
@@ -19,32 +19,38 @@ @interface FlickrPhotoMetadataDownloader()
...
+        [[NSMutableArray alloc] initWithCapacity:photosMetadata.count];
+    for (NSDictionary *data in photosMetadata) {
+      FlickrPhotoMetadata *photo =
+          [[FlickrPhotoMetadata alloc] initWithUrl:[FlickrFetcher URLforPhoto:data
+                                                                   format:FlickrPhotoFormatLarge]
```
Comment (CodeBaseCamp) [inline FlickrPhotos/FlickrPhotos/Model/Downloader/FlickrPhotoMetadataDownloader.m]: Indentation

File: FlickrPhotos/FlickrPhotos/Model/Downloader/FlickrTopPlaceDownloader.m [hunk 8]
```diff
@@ -18,34 +18,75 @@ @interface FlickrTopPlaceDownloader()
...
+      FlickrPlaceMetadata *placeMetadata =
+          [[FlickrPlaceMetadata alloc] initWithPlaceID:[place valueForKeyPath:FLICKR_PLACE_ID]
+                                               country:[self country:details]
+                                                  city:[self city:details]
+                                                region:[self region:details]];
+      NSMutableArray<FlickrPlaceMetadata *> *arrayForKey = [topPlaces objectForKey:placeMetadata.country];
+      if (!arrayForKey) {
+        arrayForKey = [[NSMutableArray alloc] init];
+        [arrayForKey addObject:placeMetadata];
+        [topPlaces setObject:arrayForKey forKey:placeMetadata.country];
```
Comment (CodeBaseCamp) [inline FlickrPhotos/FlickrPhotos/Model/Downloader/FlickrTopPlaceDownloader.m]: `[topPlaces setObject:@[arrayForKey] forKey:placeMetadata.country];` replaces lines 55-57.

File: FlickrPhotos/FlickrPhotos/Model/Downloader/FlickrTopPlaceDownloader.m [hunk 9]
```diff
@@ -18,34 +18,75 @@ @interface FlickrTopPlaceDownloader()
...
-  [self.downloader retrieveFromURL:topPlacesURL callingCompletion:extractTopPlacesData];
+  [self.downloader retrieveFromURL:[FlickrFetcher URLforTopPlaces]
+                 callingCompletion:extractTopPlacesData];
+}
+
+#pragma mark -
+#pragma mark Private methods
+#pragma mark -
+
+- (NSString *)city:(NSString *)placeInformation {
```
Comment (CodeBaseCamp) [inline FlickrPhotos/FlickrPhotos/Model/Downloader/FlickrTopPlaceDownloader.m]: `cityFromPlaceInformation:`

File: FlickrPhotos/FlickrPhotos/Model/Downloader/FlickrTopPlaceDownloader.m [hunk 10]
```diff
@@ -18,34 +18,75 @@ @interface FlickrTopPlaceDownloader()
...
+  NSArray *details = [placeInformation componentsSeparatedByString: @","] ;
+  return [[details objectAtIndex: 0]
+          stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceCharacterSet]];
+}
+
+- (NSString *)country:(NSString *)placeInformation {
```
Comment (CodeBaseCamp) [inline FlickrPhotos/FlickrPhotos/Model/Downloader/FlickrTopPlaceDownloader.m]: same

File: FlickrPhotos/FlickrPhotos/Model/Downloader/FlickrTopPlaceDownloader.m [hunk 11]
```diff
@@ -18,34 +18,75 @@ @interface FlickrTopPlaceDownloader()
...
+  NSArray *details = [placeInformation componentsSeparatedByString: @","];
+  return [[details objectAtIndex: [details count] - 1]
+          stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceCharacterSet]];
+}
+
+- (NSString *)region:(NSString *)placeInformation {
```
Comment (CodeBaseCamp) [inline FlickrPhotos/FlickrPhotos/Model/Downloader/FlickrTopPlaceDownloader.m]: same

File: FlickrPhotos/FlickrPhotos/Model/Downloader/FlickrTopPlaceDownloader.m [hunk 12]
```diff
@@ -18,34 +18,75 @@ @interface FlickrTopPlaceDownloader()
...
+                 callingCompletion:extractTopPlacesData];
+}
+
+#pragma mark -
+#pragma mark Private methods
+#pragma mark -
+
+- (NSString *)city:(NSString *)placeInformation {
+  NSArray *details = [placeInformation componentsSeparatedByString: @","] ;
+  return [[details objectAtIndex: 0]
```
Comment (CodeBaseCamp) [inline FlickrPhotos/FlickrPhotos/Model/Downloader/FlickrTopPlaceDownloader.m]: `details.firstObject`

File: FlickrPhotos/FlickrPhotos/Model/Downloader/FlickrTopPlaceDownloader.m [hunk 13]
```diff
@@ -18,34 +18,75 @@ @interface FlickrTopPlaceDownloader()
...
+          stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceCharacterSet]];
+}
+
+- (NSString *)country:(NSString *)placeInformation {
+  NSArray *details = [placeInformation componentsSeparatedByString: @","];
+  return [[details objectAtIndex: [details count] - 1]
```
Comment (CodeBaseCamp) [inline FlickrPhotos/FlickrPhotos/Model/Downloader/FlickrTopPlaceDownloader.m]: `details.lastObject`

File: FlickrPhotos/FlickrPhotos/Model/Downloader/FlickrTopPlaceDownloader.m [hunk 14]
```diff
@@ -18,34 +18,75 @@ @interface FlickrTopPlaceDownloader()
...
+          stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceCharacterSet]];
+}
+
+- (NSString *)region:(NSString *)placeInformation {
+  NSArray *details = [placeInformation componentsSeparatedByString: @","];
+  return [[details objectAtIndex: 1]
```
Comment (CodeBaseCamp) [inline FlickrPhotos/FlickrPhotos/Model/Downloader/FlickrTopPlaceDownloader.m]: `details[1]`

File: FlickrPhotos/FlickrPhotos/Model/PhotoMetadata/FlickrPhotoMetadata.h [hunk 15]
```diff
@@ -10,12 +10,26 @@ NS_ASSUME_NONNULL_BEGIN
 
 - (instancetype)init NS_UNAVAILABLE;
 
-/// Initializes with the given \c url.
-- (instancetype)initWithUrl:(NSURL *)url NS_DESIGNATED_INITIALIZER;
+/// Initializes with a given \c url, \c title and \c description.
+- (instancetype)initWithUrl:(NSURL *)url
+                      title:(NSString *)title
```
Comment (CodeBaseCamp) [inline FlickrPhotos/FlickrPhotos/Model/PhotoMetadata/FlickrPhotoMetadata.h]: Fits into line above.

File: FlickrPhotos/FlickrPhotos/Model/PhotoMetadata/FlickrPhotoMetadata.h [hunk 16]
```diff
@@ -10,12 +10,26 @@ NS_ASSUME_NONNULL_BEGIN
...
+/// Initializes instance with encoded data \c encoded.
+- (instancetype)initWithCoder:(id)encode NS_DESIGNATED_INITIALIZER;
+
+/// Encodes instance as a \c NSDictionary.
+- (NSDictionary *)serialize;
 
 /// URL of the image represented by this object.
 @property (strong, readonly, nonatomic) NSURL *url;
 
+/// Title of an image.
```
Comment (CodeBaseCamp) [inline FlickrPhotos/FlickrPhotos/Model/PhotoMetadata/FlickrPhotoMetadata.h]: `Title of the image represented by this object.`

File: FlickrPhotos/FlickrPhotos/Model/PhotoMetadata/FlickrPhotoMetadata.h [hunk 17]
```diff
@@ -10,12 +10,26 @@ NS_ASSUME_NONNULL_BEGIN
...
+@property (strong, readonly, nonatomic) NSString *title;
+
+/// Description of an image.
```
Comment (CodeBaseCamp) [inline FlickrPhotos/FlickrPhotos/Model/PhotoMetadata/FlickrPhotoMetadata.h]: `Description of the image represented by this object.`

File: FlickrPhotos/FlickrPhotos/Model/PhotoMetadata/FlickrPhotoMetadata.m [hunk 18]
```diff
@@ -7,13 +7,52 @@
 
 @implementation FlickrPhotoMetadata
 
-- (instancetype)initWithUrl:(NSURL *)url {
+- (instancetype)initWithUrl:(NSURL *)url
+                      title:(NSString *)title
+                description:(NSString *)description {
   if (self = [super init]) {
```
Comment (CodeBaseCamp) [inline FlickrPhotos/FlickrPhotos/Model/PhotoMetadata/FlickrPhotoMetadata.m]:
```
LTParameterAssert(url);
LTParameterAssert(url);
LTParameterAssert(url);
```

File: FlickrPhotos/FlickrPhotos/ImageViewController.h (hunk 3 above)
Comment (codesinc) [inline FlickrPhotos/FlickrPhotos/ImageViewController.h]: fixed

File: FlickrPhotos/FlickrPhotos/ImageViewController.h (hunk 4 above)
Comment (codesinc) [inline FlickrPhotos/FlickrPhotos/ImageViewController.h]: fixed

File: FlickrPhotos/FlickrPhotos/ImageViewController.m (hunk 5 above)
Comment (codesinc) [inline FlickrPhotos/FlickrPhotos/ImageViewController.m]: fixed

File: FlickrPhotos/FlickrPhotos/Model/PhotoMetadata/FlickrPhotoMetadata.m (hunk 18 above)
Comment (codesinc) [inline FlickrPhotos/FlickrPhotos/Model/PhotoMetadata/FlickrPhotoMetadata.m]: fixed

File: FlickrPhotos/FlickrPhotos/Model/Downloader/FlickrPhotoMetadataDownloader.m (hunk 6 above)
Comment (codesinc) [inline FlickrPhotos/FlickrPhotos/Model/Downloader/FlickrPhotoMetadataDownloader.m]: fixed

File: FlickrPhotos/FlickrPhotos/Model/Downloader/FlickrTopPlaceDownloader.m (hunk 9 above)
Comment (codesinc) [inline FlickrPhotos/FlickrPhotos/Model/Downloader/FlickrTopPlaceDownloader.m]: fixed

File: FlickrPhotos/FlickrPhotos/Model/Downloader/FlickrTopPlaceDownloader.m (hunk 10 above)
Comment (codesinc) [inline FlickrPhotos/FlickrPhotos/Model/Downloader/FlickrTopPlaceDownloader.m]: fixed

File: FlickrPhotos/FlickrPhotos/Model/Downloader/FlickrTopPlaceDownloader.m (hunk 11 above)
Comment (codesinc) [inline FlickrPhotos/FlickrPhotos/Model/Downloader/FlickrTopPlaceDownloader.m]: fixed

File: FlickrPhotos/FlickrPhotos/Model/Downloader/FlickrTopPlaceDownloader.m (hunk 13 above)
Comment (codesinc) [inline FlickrPhotos/FlickrPhotos/Model/Downloader/FlickrTopPlaceDownloader.m]: fixed

File: FlickrPhotos/FlickrPhotos/Model/Downloader/FlickrTopPlaceDownloader.m (hunk 14 above)
Comment (codesinc) [inline FlickrPhotos/FlickrPhotos/Model/Downloader/FlickrTopPlaceDownloader.m]: fixed

Comment (codesinc) [inline FlickrPhotos/FlickrPhotos/Model/Downloader/FlickrTopPlaceDownloader.m]: fixed

File: FlickrPhotos/FlickrPhotos/Model/PhotoMetadata/FlickrPhotoMetadata.h (hunk 16 above)
Comment (codesinc) [inline FlickrPhotos/FlickrPhotos/Model/PhotoMetadata/FlickrPhotoMetadata.h]: fixed

File: FlickrPhotos/FlickrPhotos/Model/PhotoMetadata/FlickrPhotoMetadata.h (hunk 17 above)
Comment (codesinc) [inline FlickrPhotos/FlickrPhotos/Model/PhotoMetadata/FlickrPhotoMetadata.h]: fixed
